### DISCUSSION TITLE
Desktop resolution displayed by client

### DISCUSSION THREAD

**Discussion by bikeking8** [LizardByte/community#119] [Sunshine]
[Best category I could find, though it's not really a config] I was getting confused with how the desktop would be displayed on the client, regardless of what video settings I chose; games looked fine but remote desktop work was real hard to see. Unless I'm incorrect and this can be improved, I did some digging and this response makes sense for those that were wondering (link below). Basically Moonlight can't affect desktop resolution, but it'll do fine with games.

https://github.com/moonlight-stream/moonlight-qt/issues/784#issuecomment-1128475228

**Comment by ReenigneArcher** [comment]
Sunshine cannot directly change the host resolution. You can do so via prep command if you'd like.

**Comment by ArrowThief** [comment]
I imagine most people know what they are doing better than me, but as lots of people are trying to figure out their new setup now that game stream is dead, here is how I solved this issue. 

My personal streaming setup involves a steam deck, so I wanted to run a 16x10 resolution and my desktop has an ultrawide monitor.
So, on connect to host I wanted to change from 3440x1440 to 2560x1600 (2x steam deck res for super sampling, because why not) and on disconnect I wanted it to switch back. 

I have not found a way to do this for direct desktop connections, but it works fine on a per application basis. 

I found a rather old bit of software called Qres that allows you to change your display resolution through command line. I downloaded that and extracted the .exe file and stuck it on my C drive. So the path was C:\Qres.exe. To use it you simply call it with "C:\Qres.exe" followed by  /x:#### and /y:#### for vertical and horizontal resolution respectively. You can also set the refresh rate with /R:###. 

In sunshine's web interface under the applications tab, when editing or creating a new application there is a partition titled "Command Preparations" under that there is a "Do" and "Undo" block.  Do will apply before the application runs and Undo will do so after you quit the application. 

My exact setup is as follows, in Do: "C:\QRes.exe /x:2560 /y:1600 /R:60" and then in Undo "C:\QRes.exe /x:3440 /y:1440 /R:100". 

If you want to copy my setup, but your host's monitor doesn't support 2560x1600, you can set a custom resolution in the Nvidia control panel.

Hope someone finds this helpful.

**Reply by GodOfWar526** [reply]
> I imagine most people know what they are doing better than me, but as lots of people are trying to figure out their new setup now that game stream is dead, here is how I solved this issue.
> 
> My personal streaming setup involves a steam deck, so I wanted to run a 16x10 resolution and my desktop has an ultrawide monitor. So, on connect to host I wanted to change from 3440x1440 to 2560x1600 (2x steam deck res for super sampling, because why not) and on disconnect I wanted it to switch back.
> 
> I have not found a way to do this for direct desktop connections, but it works fine on a per application basis.
> 
> I found a rather old bit of software called Qres that allows you to change your display resolution through command line. I downloaded that and extracted the .exe file and stuck it on my C drive. So the path was C:\Qres.exe. To use it you simply call it with "C:\Qres.exe" followed by /x:#### and /y:#### for vertical and horizontal resolution respectively. You can also set the refresh rate with /R:###.
> 
> In sunshine's web interface under the applications tab, when editing or creating a new application there is a partition titled "Command Preparations" under that there is a "Do" and "Undo" block. Do will apply before the application runs and Undo will do so after you quit the application.
> 
> My exact setup is as follows, in Do: "C:\QRes.exe /x:2560 /y:1600 /R:60" and then in Undo "C:\QRes.exe /x:3440 /y:1440 /R:100".
> 
> If you want to copy my setup, but your host's monitor doesn't support 2560x1600, you can set a custom resolution in the Nvidia control panel.
> 
> Hope someone finds this helpful.

I just wanted to say this is exactly what I was looking for, thank you!

**Reply by Infonesii** [reply]
For anyone else who followed ArrowThiefs advice i personally use MultiMonitorTool from nirsoft same concept also super old but no install required just the exe a bit simpler for those who care!

**Comment by foxy82** [comment]
I didn't have much luck with multi monitor or qres so I created my own open source app (heavily inspired by gamestream_launchpad) to do this. 

This allows me to use Playnite as a "Command" and not detached which means the session also ends when I close Playnite

You can get it here: https://github.com/foxy82/sunshine_utils/releases/tag/v0.1

Docs here: https://github.com/foxy82/sunshine_utils/blob/main/README.md

Let me know how you get on.

**Reply by ReenigneArcher** [reply]
I noticed in your readme, you're dumping the binary into the Sunshine install directory. I would avoid this, since we could remove everything in that directory with our installation.

**Reply by foxy82** [reply]
Thanks - I've changed it.

**Reply by salvitch** [reply]
Tool is fantastic works like a charm

[a comment by an automated account is not shown]
